### ISSUE TITLE
Support LNURL withdrawals

### ISSUE THREAD

**Issue by fiatjaf** [shesek/spark-wallet#104]
![demo](https://user-images.githubusercontent.com/1653275/61960649-3d512d80-af9c-11e9-97d4-b0ba00d3043d.gif)

https://github.com/btcontract/lnurl-rfc/blob/master/spec.md#3-withdrawing-funds-from-a-service

It's specially useful in these days post 0-sat invoice exploit.

It requires only that when a QR code starting with `lnurl` is found you 
1. decode it to get an `https://...` URL;
2. `GET` that URL;
3. Use the response parameters to show an invoice prompt to the user like https://lightning-wallet.com/ does -- or just use the defaults and make the invoice automatically without user interaction as t.me/lntxbot does;
4. Send the invoice back to the URL given also as a parameters.

That will generally trigger the service to pay the invoice automatically.

**Comment by fiatjaf** [comment]
I can work on that if and when you decide it's a good addition.

**Comment by Darth-Coin** [comment]
Not sure if is OK to add a comment to this, but I have a problem with LNURL Pay in Spark wallet.
In my case was this, using mobile and desktop Spark, vers 2.1.4 and c-LN 0.9..01.
Tried to pay a LNURL from for example https://lntxbot.bigsun.xyz/@Lairebyireneapolo
Scanning the QR, didn't get it.
Pasting the code, returned the error: invalid bolt11 unknown chain url

I thought that is something wrong with my Spark so I tried with other wallets (BlueWallet and BLW), both worked and paid.
If you consider that I should put this as a separate issue please let me know.

**Comment by bitpaint** [comment]
> 
> 
> Not sure if is OK to add a comment to this, but I have a problem with LNURL Pay in Spark wallet.
> In my case was this, using mobile and desktop Spark, vers 2.1.4 and c-LN 0.9..01.
> Tried to pay a LNURL from for example https://lntxbot.bigsun.xyz/@Lairebyireneapolo
> Scanning the QR, didn't get it.
> Pasting the code, returned the error: invalid bolt11 unknown chain url
> 
> I thought that is something wrong with my Spark so I tried with other wallets (BlueWallet and BLW), both worked and paid.
> If you consider that I should put this as a separate issue please let me know.

same here. spark with c-lightning don't work.
qrcode from etleneum- returned the error: invalid bolt11 unknown chain url

**Comment by shesek** [comment]
Yes, LNURL is not yet supported. I would very much like to add support though and hope to find some time to work on this in the near future. A PR would be greatly appreciated if someone's up for it. :)

(@fiatjaf sorry, I somehow missed your message about this until now. but definitely yes)
